hypershift-operator/controllers/awsendpointservice/awsendpointservice_controller.go

```
@@ -0,0 +1,208 @@
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get resource: %w", err)
+	}
+
+	// Return early if deleted
+	if !awsEndpointService.DeletionTimestamp.IsZero() {
+		completed, err := r.delete(ctx, awsEndpointService)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to delete resource: %w", err)
+		}
+		if !completed {
+			return ctrl.Result{RequeueAfter: endpointServiceDeletionRequeueDuration}, nil
+		}
+		if controllerutil.ContainsFinalizer(awsEndpointService, finalizer) {
+			controllerutil.RemoveFinalizer(awsEndpointService, finalizer)
+			if err := r.Update(ctx, awsEndpointService); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer: %w", err)
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Ensure the awsEndpointService has a finalizer for cleanup
+	if !controllerutil.ContainsFinalizer(awsEndpointService, finalizer) {
+		controllerutil.AddFinalizer(awsEndpointService, finalizer)
+		if err := r.Update(ctx, awsEndpointService); err != nil {
+			if apierrors.IsConflict(err) {
+				return ctrl.Result{Requeue: true}, nil
+			}
+			return ctrl.Result{}, fmt.Errorf("failed to add finalizer: %w", err)
+		}
+	}
+
+	// Reconcile the AWSEndpointService
+	err = reconcileAWSEndpointService(ctx, awsEndpointService, r.ec2Client, r.elbv2Client)
+	if err != nil {
+		meta.SetStatusCondition(&awsEndpointService.Status.Conditions, metav1.Condition{
+			Type:    string(hyperv1.AWSEndpointServiceAvailable),
+			Status:  metav1.ConditionFalse,
+			Reason:  hyperv1.AWSErrorReason,
+			Message: err.Error(),
+		})
+	} else {
+		meta.SetStatusCondition(&awsEndpointService.Status.Conditions, metav1.Condition{
+			Type:    string(hyperv1.AWSEndpointServiceAvailable),
+			Status:  metav1.ConditionTrue,
+			Reason:  hyperv1.AWSSuccessReason,
+			Message: "",
+		})
+	}
+	r.Status().Update(ctx, awsEndpointService)
```

why is this error being ignored?

---

```
	if err := reconcileAWSEndpointService(ctx, awsEndpointService, r.ec2Client, r.elbv2Client); err != nil {
		meta.SetStatusCondition(&awsEndpointService.Status.Conditions, metav1.Condition{
			Type:    string(hyperv1.AWSEndpointServiceAvailable),
			Status:  metav1.ConditionFalse,
			Reason:  hyperv1.AWSErrorReason,
			Message: err.Error(),
		})
	} else {
		meta.SetStatusCondition(&awsEndpointService.Status.Conditions, metav1.Condition{
			Type:    string(hyperv1.AWSEndpointServiceAvailable),
			Status:  metav1.ConditionTrue,
			Reason:  hyperv1.AWSSuccessReason,
			Message: "",
		})
		return ctrl.Result{}, r.Status().Update(ctx, awsEndpointService)
	}

	return ctrl.Result{}, r.Status().Update(ctx, awsEndpointService)
```
?

---

shouldn't be, good catch